demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -184,18 +431,40 @@ static SSL * tlsSetup( int tcpSocket )
 
         /* OpenSSL does not provide a single function for reading and loading certificates
          * from files into stores, so the file API must be called. */
-        pRootCaFile = fopen( SERVER_CERT, "r" );
+        pRootCaFile = fopen( SERVER_CERT_PATH, "r" );
 
         if( pRootCaFile != NULL )
         {
             pRootCa = PEM_read_X509( pRootCaFile, NULL, NULL, NULL );
         }
+        else
+        {
+            LogError( ( "Unable to find the certificate file in the path"
+                        " provided by SERVER_CERT_PATH(%.*s).",
+                        SERVER_CERT_PATH_LEN,
+                        SERVER_CERT_PATH ) );
+
+            /* Assert here in the step so as to help configure the path
+             * to certificate file. */
+            assert( false );
+        }
 
         if( pRootCa != NULL )
         {
             sslStatus = X509_STORE_add_cert( SSL_CTX_get_cert_store( pSslSetup ),
                                              pRootCa );
         }
+        else
+        {
+            LogError( ( "Failed to parse the server certificate from"
+                        " file %.*s. Please validate the certificate.",
+                        SERVER_CERT_PATH_LEN,
+                        SERVER_CERT_PATH ) );
+
+            /* Assert here in the step so as to help configure the path
+             * to certificate file. */
+            assert( false );
```

Comment:
I don't really have a good understanding of how asserts work under the hood. Is the file stream from the certificate automatically closed with a failed assert?

Reply:
Yea. That brings a good point. asert doesn't clean the resources properly. I think I should avoid assert in this case.